scalafix/rules/src/main/scala/fix/ClientRules.scala

```diff
@@ -0,0 +1,114 @@
+package fix
+
+import scalafix.v1._
+
+import scala.meta._
+
+sealed trait ClientType
+object ClientType {
+
+  case object Stream extends ClientType
+
+  case object Apply extends ClientType
+}
+
+object ClientRules {
+  def unapply(t: Tree)(implicit doc: SemanticDocument): Option[Patch] = t match {
+    // Client builders
+    case Importee.Name(c@Name("Http1Client")) =>
+      Some(Patch.replaceTree(c, "BlazeClientBuilder"))
+    case d@Defn.Def(_, _, _, _, _, c@Term.Apply(Term.ApplyType(Term.Name("Http1Client"), tpes), configParam)) =>
+      Some(patchClient(d, c, configParam, tpes, ClientType.Apply))
+    case d@Defn.Val(_, _, _, c@Term.Apply(Term.ApplyType(Term.Name("Http1Client"), tpes), configParam)) =>
+      Some(patchClient(d, c, configParam, tpes, ClientType.Apply))
+    case d@Defn.Var(_, _, _, c@Term.Apply(Term.ApplyType(Term.Name("Http1Client"), tpes), configParam)) =>
+      Some(patchClient(d, c, configParam, tpes, ClientType.Apply))
+    case d@Defn.Def(_, _, _, _, _, c@Term.Apply(Term.ApplyType(Term.Select(Term.Name("Http1Client"), Term.Name("stream")), tpes), configParam)) =>
+      Some(patchClient(d, c, configParam, tpes, ClientType.Stream))
+    case d@Defn.Val(_, _, _, c@Term.Apply(Term.ApplyType(Term.Select(Term.Name("Http1Client"), Term.Name("stream")), tpes), configParam)) =>
+      Some(patchClient(d, c, configParam, tpes, ClientType.Stream))
+    case d@Defn.Var(_, _, _, c@Term.Apply(Term.ApplyType(Term.Select(Term.Name("Http1Client"), Term.Name("stream")), tpes), configParam)) =>
+      Some(patchClient(d, c, configParam, tpes, ClientType.Stream))
+    case _ => None
+
+  }
+
+  private def patchClient(defn: Defn, client: Term, configParam: List[Term], tpes: List[Type], clientType: ClientType)(implicit doc: SemanticDocument) = {
+    val configParams = getClientConfigParams(configParam)
+    val ec = configParams.getOrElse("executionContext", Term.Name("global"))
+    val sslc = configParams.get("sslContext")
+    val newClientBuilder = Term.Apply(Term.ApplyType(Term.Name("BlazeClientBuilder"), tpes), List(Some(ec), sslc).flatten)
+    val withParams = withConfigParams(configParams).map(p => Patch.addRight(client, s"$p\n"))
+    Patch.replaceTree(client, newClientBuilder.toString()) ++ withParams + applyClientType(defn, client, clientType) + (ec match {
+      case Term.Name("global") =>
+        Patch.addGlobalImport(importer"scala.concurrent.ExecutionContext.Implicits.global")
+      case _ =>
+        Patch.empty
+    })
+  }
+
+
+  private[this] def applyClientType(defn: Defn, client: Tree, clientType: ClientType): Patch = clientType match {
+    case ClientType.Stream => Patch.addRight(client, ".stream")
+    case ClientType.Apply =>
```

Comment:
There is a bug here that causes a matcherror in some cases, will fix in a few hours when i get home

Reply:
Bug was in another line actually, fixed in `6503abb4a`